justl.el

```diff
@@ -60,12 +60,23 @@
 ;;
 ;; You can also control the width of the RECIPE column in the justl
 ;; buffer via `justl-recipe width`.  By default it has a value of 20.
+;;
+;; You can enable coloring of the output by adding proper hook to
+;; `compilation-filter-hook`. For Emacs older than 28.1 you need to
```

Comment:
If you consider using `compilation-filter` then I don't know how to approach that. On one hand I understand that you would rather preserve already existing functionality but on the other hand this package shouldn't add its own functions to compilation-filter-hook because it'll affect things outside of the package (like `M-x compile`). And if the package doesn't register such function itself then the user needs to discover it themselves which brings us to back to square one.

Another way would be offering customization point on the filter with the suggestion about `compilation-filter`. What do you think?

Reply:
Just to clarify the point on 28.1: as in the note `ansi-color` provides means to colorize compilation buffers but before 28.1 there was no convenient function definition so you had to define this small function yourself. I'd say it's not a bad trade-off especially when the README and doc comment provides example how to do that.

Reply:
Emacs has the variable `emacs-version`. We need to use that and switch to appropriate version of `compilation-filter-hook` based on that to support both the versions.

Let me know if you have more questions. Alternatively if you want me to work on top of your work, let me know and I can take that up.

Reply:
Yes, I'm aware of `emacs-version` but that's not the main point of my observation. The thing is I'm of the opinion that this package should never, on its own, register its own functions to `compilation-filter-hook` because this customization point is outside of the scope of the package. We can use `compilation-filter` to have unified behavior between `compile` and `justl-exec-recipe` but customization of that filter should be on the user side.

If you'd rather have this package automatically register function to colorize the output then I'd opt into either of these solutions to mimic the behaviour of `compilation-filter`:
- providing customization point on process filter (named like `justl-output-filter`)
- providing hook on predefined filter (named something like `justl-output-filter-hook`)

Since you're maintainer it's up to you to decide and I'll try to implement that.

Reply:
Is there a way to handle both - the colors as well as proper carriage return handling ?

I just checked eshell and it seems to be doing both jobs fine.

Reply:
I've pushed another approach. Let me know what you think about that.

I've tested it with `justfile`:
```
foo:
    #!/bin/bash
    printf "1/5\r"
    sleep 1
    printf "2/5\r"
    sleep 1
    printf "3/5\r"
    sleep 1
    printf "4/5\r"
    sleep 1
    printf "5/5\n"
    sleep 1
    printf "DONE\n"

bar:
    #!/bin/bash
    printf "\x1B[0mThis is normal text\n"
    printf "\x1B[31mThis is red text\n"
    printf "\x1B[32mThis is green text\n"
    printf "\x1B[34mThis is blue text\n"
    printf "\x1B[0mThis is normal text\n"
```
`foo` target:
![image](https://user-images.githubusercontent.com/32396809/176698727-049b9870-6646-4739-956e-23e59bbbb351.png)
`bar` target:
![image](https://user-images.githubusercontent.com/32396809/176698972-4db187e2-3ca0-44c4-9115-7f4d9eb5617a.png)